lib/index.js

```diff
@@ -170,8 +170,7 @@ Queue.prototype.requeue = function(item, attemptNumber, error) {
 
 Queue.prototype._enqueue = function(entry) {
   var queue = this._store.get(this.keys.QUEUE) || [];
-  queue = queue.slice(-(this.maxItems - 1));
-  queue.push(entry);
+  var removedEntrys = queue.splice(0, queue.length - (this.maxItems - 1), entry);
```

Comment:
From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

If the 2nd arg is negative or 0, there will be no items deleted

Also, Im not sure on what the minimum supported version of IE is, but Splice is IE5.5 as opposed to IE4 for Slice
They are extremely old but thought I should do my due diligence :)

Reply:
Can you add these comments to the code for future visibility?

Reply:
Will do that now :)

Reply:
Good call on this, realized past me made a mistake in reading the original implementation thinking that we deleted events at the start of the queue instead of the end.

Have updated this and added more comments :)